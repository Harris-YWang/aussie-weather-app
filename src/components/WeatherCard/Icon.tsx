import styled from 'styled-components';

const Icon = styled.img`
  height: 60px;
  display: block;
  margin: auto;
  padding: 10px;
`;

export default Icon;
