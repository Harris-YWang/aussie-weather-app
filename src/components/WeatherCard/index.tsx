import * as React from 'react';
import styled from 'styled-components';

import Card from '../Card';
import GridItem from '../GridItem';
import Icon from './Icon';
import Label from './Label';
import Title from './Title';

const Data = styled.div`
	margin: 32px 0 0 20px;
	padding-bottom: 20px;
	display: grid;
	grid-template-columns: repeat(auto-fill, minmax(140px, 1fr));
	color: #212121;
`;

const DataLabel = styled.div`
	margin: 8px 0;
`;

const Temperature = styled((props) => <div {...props} />)`
	margin: 8px 0;
	font-size: 16px;
	font-weight: 600;
	color: ${(props: any) => (props.color > 0 ? 'red' : 'blue')};
`;

const InlineBlock = styled.div`
	display: inline-block;
	margin: 0 10px 0 10px;
	vertical-align: middle;
`;

const DataValue = styled.div`
	margin: 8px 0;
	word-break: break-all;
	:after {
		content: '\\00a0';
	}
`;

const Max = styled.span`
	color: red;
`;
const Min = styled.span`
	color: blue;
`;

export interface IWeatherCardProps {
	city: string;
	weatherState?: string;
	temp: number;
	minTemp?: number;
	maxTemp?: number;
	windSpeed?: number;
	humidity?: number;
	iconUrl?: string;
}

export const WeatherCard: React.StatelessComponent<IWeatherCardProps> = ({
	city,
	weatherState,
	temp,
	minTemp,
	maxTemp,
	windSpeed,
	humidity,
	iconUrl
}) => (
	<GridItem>
		<Card>
			<Title>{city}</Title>
			<InlineBlock>
				<Icon alt={weatherState} src={iconUrl} />
				<Label>{weatherState}</Label>
			</InlineBlock>
			<InlineBlock>
				<Temperature temp={temp}>{temp} °C</Temperature>
			</InlineBlock>
			<Data>
				<DataLabel>Max</DataLabel>
				<DataValue>
					<Max>{maxTemp} °C</Max>
				</DataValue>
				<DataLabel>Min</DataLabel>
				<DataValue>
					<Min>{minTemp} °C</Min>
				</DataValue>
				<DataLabel>Wind speed</DataLabel>
				<DataValue>{windSpeed} m/s</DataValue>
				<DataLabel>Humidty</DataLabel>
				<DataValue>{humidity} %</DataValue>
			</Data>
		</Card>
	</GridItem>
);

export default WeatherCard;
