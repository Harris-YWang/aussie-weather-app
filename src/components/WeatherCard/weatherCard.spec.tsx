import { shallow } from 'enzyme';
import React from 'react';
import WeatherCard from './index';

describe('WeatherCard Component', () => {
	it('should render without crashing', () => {
		const props = {
			city: 'Sydney',
			weatherState: 'Clear',
			temp: 26.8,
			minTemp: 23.9,
			maxTemp: 30,
			windSpeed: 2.6,
			humidity: 44,
			iconUrl: 'http://openweathermap.org/img/wn/01d@2x.png'
		};
		const wrapper = shallow(<WeatherCard {...props} />);

		expect(wrapper).toMatchSnapshot();
	});
});
