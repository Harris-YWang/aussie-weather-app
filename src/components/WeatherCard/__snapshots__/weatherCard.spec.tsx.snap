// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard Component should render without crashing 1`] = `ShallowWrapper {}`;
